### PR TITLE
Shadow-ban command

### PR DESCRIPTION
**Pull request by Ascurius**
Hi JOJ0,

I quickly implemented a first appraoch to the shadow-ban command, based on your work. Hopefully this is useful to you.

If have tried to follow your documentation and command style. However, please let me know if my implementation approach does meet your requirements regarding documentation or command structure.

Additionally I added the `env` folder for virtual python environments to the `.gitignore`

### CODE REVIEW

**Comment by JOJ0** [comment]
Awesome! Please add the command to the checkbox list in readme.md.

You could also add a little more description to what it does exactly to the commands docstring which will then land up in --help. First line is what the user sees on main command -h, the rest is shown in subcommand -h

I usually steel bits and pieces from the synapse admin api docs, try to shorten. The goal is to be clear about what will happen but but not in all fancy detail like the synapse admin api docs.

**Comment by Ascurius** [comment]
I have added more description to the command (8a29b9b9923fe7fbb25af93fe2fab45a90a3a47e) and also added it to the command checkbox list in the README.md (3d27354530566a9d72086dcc26ea0151fc970432)

File: synadm/api.py [hunk 1]
```diff
@@ -1063,3 +1063,16 @@ def regtok_delete(self, token):
 
         """
         return self.query("delete", f"v1/registration_tokens/{token}")
+
+    def user_shadow_ban(self, user_id, unban):
+        """ Shadow-ban or unban a user
+
+        Args:
+            user_id (string): The user to be baned/unbanned
```
Comment (JOJ0) [inline synadm/api.py:1071]: Typo: banned

File: synadm/api.py [hunk 2]
```diff
@@ -1063,3 +1063,16 @@ def regtok_delete(self, token):
...
+            unban (boolean): Unban the specified user
```
Comment (JOJ0) [inline synadm/api.py:1072]: End "sentences" with dot.

File: synadm/cli/user.py [hunk 3]
```diff
@@ -554,3 +554,24 @@ def user_login_cmd(helper, user_id, expire_days, expire, expire_ts,
...
+    Useful for moderating malicious or egregiously abusive users.
+
+    A shadow-banned user will not receive any notification, but still will be
+    unable to contact anyone on the server. Use this as a tool of last resort since it
+    may lead to confusing or broken behaviour for the client. 
+    
+    Generally, it is more appropriate to ban or kick abusive users.
+    """
+    user_ban = helper.api.user_shadow_ban(user_id, unban)
+    helper.output(user_ban)
```
Comment (JOJ0) [inline synadm/cli/user.py:577]: This API seems to always return empty dict {} when no error occured. If you want but not strictly required you could prettify that for human mode. Something like this: https://github.com/Ascurius/synadm/blob/master/synadm/cli/user.py#L127-L134

File: synadm/cli/user.py [hunk 4]
```diff
@@ -554,3 +554,24 @@ def user_login_cmd(helper, user_id, expire_days, expire, expire_ts,
...
+)
+@click.pass_obj
+def user_shadow_ban(helper, user_id, unban):
+    """Shadow-ban or unban a user.
+
+    Useful for moderating malicious or egregiously abusive users.
+
+    A shadow-banned user will not receive any notification, but still will be
+    unable to contact anyone on the server. Use this as a tool of last resort since it
+    may lead to confusing or broken behaviour for the client.
```
Comment (JOJ0) [inline synadm/cli/user.py:572]: Please keep the lines 79 characters max. Don't know what editor you are using but with something `vim`'ish the quickest way would be marking the lines and pressing `gq`

File: synadm/cli/user.py [hunk 5]
```diff
@@ -554,3 +554,24 @@ def user_login_cmd(helper, user_id, expire_days, expire, expire_ts,
...
+def user_shadow_ban(helper, user_id, unban):
+    """Shadow-ban or unban a user.
+
+    Useful for moderating malicious or egregiously abusive users.
+
+    A shadow-banned user will not receive any notification, but still will be
+    unable to contact anyone on the server. Use this as a tool of last resort since it
+    may lead to confusing or broken behaviour for the client. 
+    
+    Generally, it is more appropriate to ban or kick abusive users.
```
Comment (JOJ0) [inline synadm/cli/user.py:574]: There is some redundant spaces.

File: synadm/cli/user.py (hunk 4 above)
Comment (JOJ0) [reply synadm/cli/user.py:572]: Also there is a redundant space after client.

**Review by JOJ0 — CHANGES_REQUESTED** [review]
Sorry for the delay. :-) A view cosmetical changes, besides that LGTM and ready to merge :-)

**Comment by Ascurius** [comment]
Sorry for my late response. I modified the section you mentioned according to your requirements. Besides, I have added some additional body to the shadow-ban command for the human mode. Please let me know if anything else needs to be changed.

**Comment by JOJ0** [comment]
No worries, all good, thanks a lot for the changes. Merged and happy synadm got a nice new feature. Many thanks!

**Comment by JOJ0** [comment]
BTW I'd like to contact you because of something else and invite you to synadm:peek-a-boo.at

**Pull request merged**